sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/specialized/BlobAsyncClientBase.java

```diff
@@ -1384,4 +1404,44 @@ private void downloadToFileCleanup(AsynchronousFileChannel channel, String fileP
                 return new SimpleResponse<>(rb, new StorageAccountInfo(hd.getSkuName(), hd.getAccountKind()));
             });
     }
+
+    /**
+     * Generates a user delegation sas for the blob using the specified
+     * {@link BlobServiceSasSignatureValues}.
+     * @see BlobServiceSasSignatureValues for more information on how to construct a user delegation SAS.
+     *
+     * <p><strong>Code Samples</strong></p>
+     *
+     * {@codesnippet com.azure.storage.blob.specialized.BlobAsyncClientBase.generateUserDelegationSas#BlobServiceSasSignatureValues-UserDelegationKey}
+     *
+     * @param blobServiceSasSignatureValues {@link BlobServiceSasSignatureValues}
+     * @param userDelegationKey {@link UserDelegationKey}
+     *
+     * @return A {@code String} representing all SAS query parameters.
+     */
+    public String generateUserDelegationSas(BlobServiceSasSignatureValues blobServiceSasSignatureValues,
+        UserDelegationKey userDelegationKey) {
+        return getBlobAsyncClient().generateUserDelegationSas(blobServiceSasSignatureValues, userDelegationKey);
```

Comment:
Shouldn't the implementation of this method and `generateSas` live in this class instead of `BlobAsyncClient` given that `BlobAsyncClientBase` is the super class to `BlobAsyncClient`? This goes against common practice by having the super class using the sub-classes implementation.

Additionally moving the implementation here means `AppendBlobAsyncClient`, `BlockBlobAsyncClient`, and `PageBlobAsyncClient` get the implementation for free.

Reply:
Additionally this will allow for the removal of `prepareBuilder` and `getBlobAsyncClient`.

Reply:
Yep, I just moved this in an earlier commit since Rick mentioned it. I forgot that BlobAsyncClient inherited from BlobAsyncClientBase